internal/circle/errors.go

```diff
@@ -0,0 +1,54 @@
+package circle
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+)
+
+// APIError represents the error response from Circle APIs.
+type APIError struct {
+	Code    int              `json:"code"`
+	Message string           `json:"message"`
+	Errors  []APIErrorDetail `json:"errors,omitempty"`
+}
+
+// APIErrorDetail represents the detailed error information.
+type APIErrorDetail struct {
+	Error        string                 `json:"error"`
+	Message      string                 `json:"message"`
+	Location     string                 `json:"location"`
+	InvalidValue interface{}            `json:"invalidValue,omitempty"`
+	Constraints  map[string]interface{} `json:"constraints,omitempty"`
+}
+
+// Error implements the error interface for APIError.
+func (e APIError) Error() string {
+	return fmt.Sprintf("APIError: Code=%d, Message=%s, Errors=%v", e.Code, e.Message, e.Errors)
+}
+
+// parseAPIError parses the error response from Circle APIs.
+// https://developers.circle.com/circle-mint/docs/circle-apis-api-errors.
+func parseAPIError(resp *http.Response) (*APIError, error) {
+	if resp.StatusCode == http.StatusUnauthorized {
+		return nil, fmt.Errorf("unauthorized to access the resource")
+	}
+
+	if resp.StatusCode == http.StatusForbidden {
+		return nil, fmt.Errorf("forbidden to access the resource")
+	}
```

Comment:
👋  I think these responses should trigger an API error, so the caller can identify the 401/403 issue, and decide what to do with it. WDYT?

Reply:
So `APIError` here is a Circle resource that's returned (mostly with a 400 http status) and that signifies there is a user error with the API. The code in the APIError is not the http status code, but a Circle specific code as described here https://developers.circle.com/circle-mint/docs/circle-apis-api-errors 

Given that 401 and 403 issues prevent us from even fetching the APIError, I handled them differently. I'll try to think of a way to propagate these properly.

Reply:
Hum, when I executed the request with an invalid key I did get the API error with the 401 status code:

```curl
curl --location 'https://api-sandbox.circle.com/v1/transfers' \
--header 'Content-Type: application/json' \
--header 'Authorization: Bearer aaa' \
--write-out '\nHTTP Status: %{http_code}\n------\n'
```

Returned:
```text
{"code":401,"message":"Malformed key. Does it contain three parts?"}
HTTP Status: 401
```